onnxruntime/core/util/math_cpu.cc

```diff
@@ -353,7 +378,8 @@ void Im2col<T, StorageOrder::NCHW>::operator()(
         }
       }
     }
-  }
+  };
+  ThreadPool::TryBatchParallelFor(tp, channels, channel_proc, batches);
```

Comment:
TryBatchParallelFor assigns a fixed workload to each thread. Will TryParallelFor with a tuned cost better?

Reply:
I am not sure. I looked at the algo there and it seems to assume that we are using all the available CPUs  and I think this is not desirable. We can try both.